swap/htlc.go

```diff
@@ -512,3 +561,227 @@ func (h *HtlcScriptV2) MaxTimeoutWitnessSize() int {
 func (h *HtlcScriptV2) SuccessSequence() uint32 {
 	return 1
 }
+
+// HtlcScriptV3 encapsulates the htlc v3 script.
+type HtlcScriptV3 struct {
+	TimeoutScript  []byte
+	ClaimScript    []byte
+	TaprootKey     []byte
+	InternalPubKey []byte
+}
+
+// newHTLCScriptV3 constructs a HtlcScipt with the HTLC V3 taproot script.
+func newHTLCScriptV3(cltvExpiry int32, receiverHtlcKey, senderHtlcKey [33]byte,
+	swapHash lntypes.Hash) (*HtlcScriptV3, error) {
+
+	_, receiverPubKey := btcec.PrivKeyFromBytes(
```

Comment:
Pubkey from privkeyfrombytes? 


```
receiverPubkey, err:= btcec.ParsePubkey(receiverHtlcKey[:])
copy(schnorrReceiverKey, schnorr.SerializePubkey(receiverPubkey)
```

Reply:
Why can't we just store `[33]byte` then `schnorr.SerializePubkey` when we need to use the key in a `HtlcV3` context? 

Do you have a link to that convo so I can understand the context of that discussion?